sdks/java/core/src/test/java/org/apache/beam/sdk/testing/BigqueryMatcherTest.java

```diff
@@ -144,7 +131,27 @@ public void testQueryWithRetriesWhenServiceFails() throws Exception {
         new BigqueryMatcher(appName, projectId, query, "some-checksum"));
     when(mockQuery.execute()).thenThrow(new IOException());
 
-    thrown.expect(IOException.class);
+    thrown.expect(RuntimeException.class);
+    thrown.expectMessage("Unable to get BigQuery response after retrying");
+
+    matcher.queryWithRetries(
+        mockBigqueryClient,
+        new QueryRequest(),
+        fastClock,
+        BigqueryMatcher.BACKOFF_FACTORY.backoff());
+
+    verify(matcher).newBigqueryClient(eq(appName));
```

Comment:
well. since try block is already used, having `thrown.expect()` is a bit duplicate to me. One reason I prefer:
```
try { 
  func.call()
} catch (Exception e) {
  assertThat()
  verify()
}
fail()
```
is that all verification steps (`asserThat` and `verify`) can be placed in one block which looks more readable.